Update what-best-practices-are-recommended-for-naming-metrics-and-tag… (#21515)
DataDog/documentation: content/en/developers/guide/what-best-practices-are-recommended-for-naming-metrics-and-tags.md

  - /developers/faq/what-best-practices-are-recommended-for-naming-metrics-and-tags
---

> Naming convention is an art and possibly one of the most difficult decisions to agree on. Defining a naming convention for your metrics, tags and services is crucial to have a clean, readable and maintanable telemetry data. Here are some recommendations:

Reviewer: ```suggestion
Naming convention is an art and possibly one of the most difficult decisions to agree on. Defining a naming convention for your metrics, tags and services is crucial to have a clean, readable, and maintainable telemetry data. Here are some recommendations:
```

* Provide descriptive and meaningful names: the metrics or tags clearly describe the purpose or meaning of the value.

> * Adhere to the format and limitations depicted below.

Reviewer: ```suggestion
* Adhere to the format and limitations described below.
```

> * Avoid abbreviations that might be overloeaded with multiple meanings

Reviewer: ```suggestion
* Avoid abbreviations that might have multiple meanings
```

> * Maintain consistency across all teams, apps, services.

Reviewer: ```suggestion
* Maintain consistency across all teams, apps, and services.
```

* Avoid reserved keywords that might cause clashes with the other tags or metrics.

> * In the case of metrics, prefix them with a namespace depicting the application or service the data is generated from

Reviewer: ```suggestion
* In the case of metrics, prefix them with a namespace depicting the application or service generating the data.
```
